cal/cal/static/js/components/category-details.js

```diff
@@ -63,61 +61,34 @@ function CategoriesDetailCtrl($scope, $http, QueryService){
   }.bind(this);
 
   this.showDaily = function() {
-    $http({
-      method: 'GET',
-      url: this.timeseriesDay + '.json',
-      params: {
-        timezone: query_timezone,
-      }
-    }).
-    then(function successCallback(response) {
+    QueryService.populateDay('Category day' + _this.categoryId, 'Category', _this.categoryId, []).
+    then(function populate(data) {
       _this.timeStep = "day";
-      _this.averageHours = Math.round(((_this.categoryHours / response.data[0].length) * 100)) / 100;
-      var eventData = QueryService.populateData(response.data, 'Category');
-      _this.ctrlGraphData = eventData[0];
-      _this.showGraph(eventData[1]);
+      _this.averageHours = Math.round(((_this.categoryHours / data[0][0].values.length) * 100)) / 100;
+      _this.ctrlGraphData = data[0];
```

Comment:
Sorry to introduce this kinda late but could you make the `[0]` and `[1]` keyworded dicts instead of referencing an index in a tuple? If we reorder anything in the backend we will unknowingly screw up our frontend code.